tests/unit/lms/services/d2l_api/client_test.py

```diff
@@ -3,7 +3,73 @@
 import pytest
 
 from lms.services.d2l_api._basic import BasicClient
-from lms.services.d2l_api.client import D2LAPIClient
+from lms.services.d2l_api.client import D2LAPIClient, D2LTableOfContentsSchema
+from tests import factories
+
+
+class TestD2LTableOfContentsSchema:
+    def test_valid(self, table_of_contents_response):
+        schema = D2LTableOfContentsSchema(
+            factories.requests.Response(json_data=table_of_contents_response)
+        )
+
+        result = schema.parse()
+
+        assert result == {
+            "modules": [
+                {
+                    "topics": [
+                        {
+                            "id": "ID 1",
+                            "name": "Title 1",
+                            "updated_at": "Date 1",
+                            "type": "File",
+                        }
+                    ],
+                    "modules": [
+                        {
+                            "topics": [
+                                {
+                                    "id": "ID 1",
+                                    "name": "Title 1",
+                                    "updated_at": "Date 1",
+                                    "type": "File",
+                                }
+                            ],
+                        }
+                    ],
+                }
+            ]
+        }
+
+    @pytest.fixture
+    def table_of_contents_response(self):
+        return {
+            "Modules": [
+                {
+                    "Topics": [
+                        {
+                            "Identifier": "ID 1",
+                            "Title": "Title 1",
+                            "LastModifiedDate": "Date 1",
+                            "TypeIdentifier": "File",
+                        }
+                    ],
+                    "Modules": [
+                        {
+                            "Topics": [
+                                {
+                                    "Identifier": "ID 1",
+                                    "Title": "Title 1",
+                                    "LastModifiedDate": "Date 1",
+                                    "TypeIdentifier": "File",
```

Comment:
```suggestion
                                    "Identifier": "ID 2",
                                    "Title": "Title 2",
                                    "LastModifiedDate": "Date 2",
                                    "TypeIdentifier": "File",
```

Might as well take advantage to test we don't scramble the results?